clang/test/CodeGen/cheri/convert-int-to-cap.c

```diff
@@ -106,21 +106,6 @@ char *__capability test_ptr_to_capptr_tocap(char *p) {
   // AST-NEXT:  ImplicitCastExpr {{.+}} 'char *' <LValueToRValue> part_of_explicit_cast
 }
 
-// CHECK-LABEL: define {{[^@]+}}@test_ptr_to_capptr_via_addr_intcap
-// CHECK-SAME: (i8* [[P:%.*]]) #0
-// CHECK-NEXT:  entry:
-// CHECK-NEXT:    [[TMP0:%.*]] = ptrtoint i8* [[P]] to i64
-// CHECK-NEXT:    [[TMP1:%.*]] = getelementptr i8, i8 addrspace(200)* null, i64 [[TMP0]]
-// CHECK-NEXT:    ret i8 addrspace(200)* [[TMP1]]
-//
-char *__capability test_ptr_to_capptr_via_addr_intcap(char *p) {
-  return (char *__capability)(__cheri_addr __intcap)p;
```

Comment:
Interesting, so we were actually already doing the provenance-stripping cast here. But not allowing it avoids the broken case for globals, so that is definitely better.

Reply:
For hybrid, because that's the same behaviour as `(char *__capability)(__intcap)p` gives when `p` is an integer pointer; that case was never broken, but was a no-op because you weren't operating on a capability in the first place.